Source/WebCore/html/HTMLAttachmentElement.cpp

```diff
@@ -147,27 +158,56 @@ void HTMLAttachmentElement::ensureModernShadowTree(ShadowRoot& root)
     m_innerLegacyAttachment->m_icon = WTFMove(m_icon);
     m_innerLegacyAttachment->m_iconSize = m_iconSize;
     m_innerLegacyAttachment->setIdAttribute(attachmentPreviewIdentifier());
-    container->appendChild(*m_innerLegacyAttachment);
+    m_elementContainer->appendChild(*m_innerLegacyAttachment);
 
     m_elementWithAction = HTMLDivElement::create(document());
     m_elementWithAction->setIdAttribute(attachmentActionIdentifier());
     if (const auto& action = attachmentActionForDisplay(); !action.isEmpty())
         m_elementWithAction->setInnerText(String { action });
-    container->appendChild(*m_elementWithAction);
+    m_elementContainer->appendChild(*m_elementWithAction);
 
     m_elementWithTitle = HTMLDivElement::create(document());
     m_elementWithTitle->setIdAttribute(attachmentTitleIdentifier());
     if (auto title = attachmentTitleForDisplay(); !title.isEmpty())
         m_elementWithTitle->setInnerText(WTFMove(title));
-    container->appendChild(*m_elementWithTitle);
+    m_elementContainer->appendChild(*m_elementWithTitle);
 
     m_elementWithSubtitle = HTMLDivElement::create(document());
     m_elementWithSubtitle->setIdAttribute(attachmentSubtitleIdentifier());
     if (auto subtitle = attachmentSubtitleForDisplay(); !subtitle.isEmpty())
         m_elementWithSubtitle->setInnerText(WTFMove(subtitle));
-    container->appendChild(*m_elementWithSubtitle);
+    m_elementContainer->appendChild(*m_elementWithSubtitle);
+
```

Comment:
fwiw, instead of multiple appendChild calls, you can do `m_elementContainer->append(X, Y, Z)` which appends everything at once.

Reply:
I don't see that function.
There's `ContainerNode::append(FixedVector<NodeOrString>&& vector)`, it doesn't seem used a lot, and it would do a lot of work to construct this `FixedVector`, then convert it to a small tree of nodes, and that would append that tree to the container. If that's what you were thinking of, are you sure that's better?

Reply:
Yeah I was thinking of this function, though I realize this is C++ so it's not as fancy as when you call it from JS 😅 

Feel free to ignore this.

Reply:
Thanks. But you made me notice the repetitions; I'll see if I can refactor these similar-looking blocks...

Reply:
I've refactored this whole section, please have a look.